narwhals/dataframe.py

```diff
@@ -3109,13 +3110,25 @@ def lazy(self: Self) -> Self:
     def gather_every(self: Self, n: int, offset: int = 0) -> Self:
         r"""Take every nth row in the DataFrame and return as a new DataFrame.
 
+        !!! warning
+            `LazyFrame.gather_every` is deprecated and will be removed in a future version.
+            Note: this will remain available in `narwhals.stable.v1`.
+            See [stable api](../backcompat.md/) for more information.
+
         Arguments:
             n: Gather every *n*-th row.
             offset: Starting index.
 
         Returns:
             The LazyFrame containing only the selected rows.
         """
+        msg = (
+            "`LazyFrane.gather_every` is deprecated and will be removed in a future version.\n\n"
+            "Note: this will remain available in `narwhals.stable.v1`.\n"
+            "See https://narwhals-dev.github.io/narwhals/backcompat/ for more information.\n"
+        )
+        issue_deprecation_warning(msg, _version="1.28.0")
```

Comment:
1.29?

Reply:
I checked how it was used in the other cases e.g. in `Expr.tail` https://github.com/narwhals-dev/narwhals/blob/77a0150ffb5121086d74b0f9a10ec26781162529/narwhals/expr.py#L1784

and I noticed that it corresponds to the current version (at the time) and in the docs we mention the release right after. I am not sure if this was intentional or not - the description for `_version` made it ambiguous to me:

```py
_version: Narwhals version when the warning was introduced. Just used for internal
            bookkeeping.
```

Reply:
ah sorry 😳 this is just internal anyway so as long as it's approximately right it's probably fine 😄

Reply:
if we want to be consistent going forwards, i think the intention was probably for it to be the next minor release after the current one

if it ends up in a patch release in between then no big deal

Reply:
Should I change the other instances in which it is used to match what's in the doc?